app/main.py

```diff
@@ -1,3 +1,48 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, int]) -> Union[Distance, int]:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        self.km = self.km + other
+        return self
```

Comment:
The `__add__` method should return a new `Distance` instance when adding an integer, similar to when adding another `Distance` object. Modify the method to return `Distance(self.km + other)` instead of modifying `self.km` directly.